### DISCUSSION TITLE
destination mapping

### DISCUSSION THREAD

**Discussion by Nielkrokodil** [streetcomplete/StreetComplete#5035] [Ideas]
I think Streetcomplete could boost destination mapping (https://wiki.openstreetmap.org/wiki/Key:destination). But it might be hard to implement. 

On intersections of two high priority roads (motorway, trunk, primary, secondary, maybe even tertiary) there almost always are such signs, but sich signs can be found on other highway classes as well.

When driving it really helps to know which destination to look for, especially in roundabouts so there is a big benefit if it is mapped. 

Potential problems with this task could be:
- the way continues through the intersection. Solution: split way at intersection
- the way leaving the intersection continues past the next intersection and would result in wrong mapping there. Solution: Split way at next intersection
- the way is very long. After a potential split the destination tag would be duplicated and far from the intersection, resulting in bad mapping. Solution: Split the way after a specified distance (30m?) if longer than another specified distance (100m?).
- destination signs often have multiple tags (destination, destination:ref, destination symbol etc). Solution: if quest is regular quest: only ask for destination and add destination:signed=no if no signs are posted. If quest is overlay quest: ask for destination and some other tags.
- "destination varies by lane" might be too complex for the app. Solution: I don't know. How often is this the case?
- the in-app-shown sign does not match the country specific sign (color, text style etc) and by that causes confusion: Solution: Add country specific designs and/or provide a good introduction explaining the differences. 

What do you think? Especially should it be a regular task or an overlay task.

**Comment by riQQ** [comment]
Related: #3806 and for SCEE Helium314/SCEE#419

**Reply by mnalis** [reply]
The main problem seem that it is very complex (both technically to implement, but even more so for the users - note the the target audience for SC are beginners) so it would be very like wrong / misleading data is entered. Probably the best would be to just long press to add a note with a picture on places missing the destination, and then add it later from home in some full-fledged editor.

Or use more advanced mobile editor like [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci) which allows for all those considerations and edits in the first place.

**Reply by Nielkrokodil** [reply]
I know SC is good for beginners, but many experienced mappers use it also. Some difficult tasks are hidden per default to reduce errors by beginners but to let experienced people use it. I would see such a destination to be hidden by default.

To the technical problems: I think when it is done as overlay it really is complicated because it has to handle destination, destination:lanes and relations. If it is a regular quest, it could simply (?) skip all sites with any destination mapping and to the rest add the destination-tag to the way. Only the way-splits need to be handled in this case.

**Reply by riQQ** [reply]
There's also some discussion in the SCEE (StreetComplete Expert Edition): Helium314/SCEE#419.
